pkg/devices/gen_net.go

```diff
@@ -31,29 +33,52 @@ type GenNetDevice struct {
 	ifName    string
 	linkType  string
 	linkSpeed string
-	vfID      int
+	funcID    int
 	isRdma    bool
 }
 
 // NewGenNetDevice returns GenNetDevice instance
-func NewGenNetDevice(dev *ghw.PCIDevice, isRdma bool) (*GenNetDevice, error) {
-	var ifName string
-	pfName, err := utils.GetPfName(dev.Address)
-	if err != nil {
-		glog.Warningf("unable to get PF name %q", err.Error())
-	}
-
-	netDevs, _ := utils.GetNetNames(dev.Address)
-	if len(netDevs) == 0 {
-		ifName = ""
-	} else {
-		ifName = netDevs[0]
+func NewGenNetDevice(deviceID string, dt types.DeviceType, isRdma bool) (*GenNetDevice, error) {
+	var netNames []string
+	var pfName string
+	var pfAddr string
+	var funcID int
+	var err error
+
+	//nolint: exhaustive
```

Comment:
same question as above.

Reply:
Used it to make `accelerator` case go to `default`. If linter tag is undesired I will add case explicitly.